sdk/dapp-kit/src/themes/themeContract.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { createGlobalThemeContract } from '@vanilla-extract/css';
+
+const themeContractValues = {
+	blurs: {
+		modalOverlay: '',
+	},
+	backgroundColors: {
+		primaryButton: '',
+		primaryButtonHover: '',
+		outlineButtonHover: '',
+		walletItemHover: '',
+		walletItemSelected: '',
+		modalOverlay: '',
+		modalPrimary: '',
+		modalSecondary: '',
+		iconButton: '',
+		iconButtonHover: '',
+		dropdownMenu: '',
+		dropdownMenuSeparator: '',
+	},
+	borderColors: {
+		outlineButton: '',
+	},
+	colors: {
+		primaryButton: '',
+		outlineButton: '',
+		body: '',
+		bodyMuted: '',
+		bodyDanger: '',
+		iconButton: '',
+	},
+	radii: {
+		small: '',
+		medium: '',
+		large: '',
+		xlarge: '',
+	},
+	shadows: {
+		primaryButton: '',
+		walletItemSelected: '',
+	},
+	fontWeights: {
+		normal: '',
+		medium: '',
+		bold: '',
+	},
+	typography: {
+		fontFamily: '',
+		fontSize: '',
+		fontStyle: '',
+		lineHeight: '',
+		letterSpacing: '',
+	},
+};
+
+export type ThemeVars = typeof themeContractValues;
```

Comment:
what if we accepts themes as something like:

`themes: { mediaQuery?: string; selectorPrefix?: string; variables: ThemeVars }[]`

This would allow you to pass in a light and dark theme with separate queries, or have sparate mobile/desktop themes, or use selectorPrefix to change theme based on a classname.  Would be pretty easy to apply that format with the injector logic you already have.

Reply:
it also get's away from the slightly janky detection `theme.light ? theme.light : theme`

Reply:
we could do `ThemeVariables | ThemeList` so you can handle still accept `theme={lightTheme}` for the simple case

Reply:
we can also export a pre-configured array with the light/dark themes which should cover a big chunk of the use cases, or do something like. `theme={autoTheme({ light: lightTheme, dark: dartTheme })}`

Reply:
I left a comment elsewhere but I think it's pretty common to do theming outside of media queries.

Reply:
yeah, that's why I was trying to suggest something that allows either a prefix or a media query (or both).  using a class on body is pretty common from what I remember

Reply:
Okay, I ran with @hayes-mysten suggestion and I think this works pretty nicely. And yep, we definitely want to be able to support non-media-query theme selectors as y'all mentioned above (https://github.com/pacocoursey/next-themes is one of the more popular libraries for managing themes in a SSR-supported env)

Static theme:
```
<WalletProvider theme={lightTheme}>abc</WalletProvider>
```

Dynamic theme based on a single selector:
```
		<WalletProvider
			theme={[
				{
					selector: '.light',
					variables: lightTheme,
				},
			]}
		>
			abc
		</WalletProvider>
```

Multiple custom themes with various selectors:
```
		<WalletProvider
			theme={[
				{
					mediaQuery: '(prefers-color-scheme: dark)',
					selector: '.light',
					variables: lightTheme,
				},
				{
					selector: '.purple',
					variables: lightTheme,
				},
			]}
		>
			abc
		</WalletProvider>
```